engine/player/player.cpp

```diff
@@ -3029,6 +3029,123 @@ void player_t::init_finished()
       }
     } );
   }
+
+  if ( !precombat_state_map.empty() )
+  {
+    sim->error( "Warning: The 'override.precombat_state' option may not be fully supported for all buffs and cooldowns and and may produce incorrect or misleading results." );
+    std::unordered_map<buff_t*, std::tuple<int, double, timespan_t>> precombat_buff_state;
```

Comment:
Instead of using a tuple, wouldn't it make the code clearer and easier to read if you defined a struct for it?
```
struct buff_state_t{
int stacks;
double value;
timespan_t duration
};
std::unordered_map<buff_t*,buff_state_t> precombat_buff_state;
```

Reply:
Yeah, a struct is probably a bit better. It works with default values as well, which is nice.